### PR TITLE
Cluster config api return verbose PA status

### PR DESCRIPTION
**Pull request by dzane17**
Signed-off-by: David Zane <davizane@amazon.com>

**Is your feature request related to a problem? Please provide an existing Issue # , or describe.**
Fixes https://github.com/opensearch-project/performance-analyzer/issues/322

When checking cluster state of Performance Analyzer, the responses are bit-encoded numeric values, which don't provide much meaningful insight.

Current cluster/config output:
```
GET localhost:9200/_plugins/_performanceanalyzer/cluster/config
{"batchMetricsRetentionPeriodMinutes": 7, "currentPerformanceAnalyzerClusterState": 19, "shardsPerCollection": 0}
```

**Describe the solution you are proposing**
Translate bitmap to Map<String, Boolean> output when ?verbose query param is provided.

New cluster/config output:
```
% curl -XGET localhost:9200/_plugins/_performanceanalyzer/cluster/config?verbose
{"currentPerformanceAnalyzerClusterState":{"PerformanceAnalyzerEnabled":true,"RcaEnabled":true,"LoggingEnabled":false,"BatchMetricsEnabled":false,"ThreadContentionMonitoringEnabled":false},"shardsPerCollection":0,"batchMetricsRetentionPeriodMinutes":7}
```

Also works for these GET/POST APIs:
```
/cluster/config?verbose
/rca/cluster/config?verbose
/logging/cluster/config?verbose
/batch/cluster/config?verbose
/threadContentionMonitoring/cluster/config?verbose
```

**Describe alternatives you've considered**
I have kept bitmap implementation instead of converting to some map object because of the provided [Setting](https://github.com/opensearch-project/OpenSearch/blob/913cb3a9ec74d46c4a0840fcc35cd1cbe3c326ac/server/src/main/java/org/opensearch/common/settings/Setting.java#L1357
) construct. We are using `Setting.intSetting()`. There is no map option (int, float, long, boolean... available).

Also, int representation will help a bit with memory.

**Additional context**

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/performance-analyzer/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/org/opensearch/performanceanalyzer/config/setting/handler/PerformanceAnalyzerClusterSettingHandler.java [hunk 1]
```diff
@@ -6,6 +6,7 @@
 package org.opensearch.performanceanalyzer.config.setting.handler;
 
 
+import java.util.*;
```
Comment (kiranprakash154) [inline src/main/java/org/opensearch/performanceanalyzer/config/setting/handler/PerformanceAnalyzerClusterSettingHandler.java:9]: lets import only the required libraries.

File: src/test/java/org/opensearch/performanceanalyzer/config/setting/handler/PerformanceAnalyzerClusterSettingHandlerTests.java [hunk 2]
```diff
@@ -9,6 +9,7 @@
 import static org.mockito.MockitoAnnotations.initMocks;
 import static org.powermock.api.mockito.PowerMockito.when;
 
+import java.util.*;
```
Comment (kiranprakash154) [inline src/test/java/org/opensearch/performanceanalyzer/config/setting/handler/PerformanceAnalyzerClusterSettingHandlerTests.java:12]: same

File: src/test/java/org/opensearch/performanceanalyzer/config/setting/handler/PerformanceAnalyzerClusterSettingHandlerTests.java [hunk 3]
```diff
@@ -56,7 +67,14 @@ public void paDisabledClusterStateTest() {
         clusterSettingHandler =
                 new PerformanceAnalyzerClusterSettingHandler(
                         mockPerformanceAnalyzerController, mockClusterSettingsManager);
-        assertEquals(0, clusterSettingHandler.getCurrentClusterSettingValue());
+        Map<String, Boolean> statusMap =
+                createStatusMap(
+                        DISABLED_STATE,
```
Comment (kiranprakash154) [inline src/test/java/org/opensearch/performanceanalyzer/config/setting/handler/PerformanceAnalyzerClusterSettingHandlerTests.java:72]: I see this statusMap of all disabled states being used twice in the same file, lets declare it once and re-use with better naming (since statusMap does not tell mnuch about the contents), probably cluster_settings_fully_disabled, cluster_settings_fully_enabled, cluster_settings_partly_enabled ? feel free to go with other better names.

File: src/main/java/org/opensearch/performanceanalyzer/config/setting/handler/PerformanceAnalyzerClusterSettingHandler.java (hunk 1 above)
Comment (kiranprakash154) [reply src/main/java/org/opensearch/performanceanalyzer/config/setting/handler/PerformanceAnalyzerClusterSettingHandler.java:9]: This could be your IDE optimizing imports on your behalf.

File: src/main/java/org/opensearch/performanceanalyzer/config/setting/handler/PerformanceAnalyzerClusterSettingHandler.java [hunk 4]
```diff
@@ -132,8 +140,20 @@ public void onSettingUpdate(final Integer newSettingValue) {
      *
      * @return the current cluster setting value if exists. Initial cluster setting otherwise.
      */
-    public int getCurrentClusterSettingValue() {
-        return currentClusterSetting;
+    public Map<String, Boolean> getCurrentClusterSettingValue() {
```
Comment (kiranprakash154) [inline src/main/java/org/opensearch/performanceanalyzer/config/setting/handler/PerformanceAnalyzerClusterSettingHandler.java:143]: How is the map being used by the caller now that we changed the structure of the response? Shouldn't we change that as well ?

**Comment by sgup432** [comment]
As this is a breaking change(as api response is essentially changing), I think we should instead introduce a new parameter like ```?expand``` and return meaningful status only when this is used. Using this we won't essentially need to fix [integ tests](https://github.com/opensearch-project/performance-analyzer/blob/3ab57eb8802329ac8a11e9fd3e1dbd605c8a35e7/src/test/java/org/opensearch/performanceanalyzer/PerformanceAnalyzerIntegTestBase.java#L262) or make change changes in RCA.

We can just update the README for this param usage. @kiranprakash154 @dzane17 Thoughts?

```
GET localhost:9200/_plugins/_performanceanalyzer/cluster/config
{"batchMetricsRetentionPeriodMinutes": 7, "currentPerformanceAnalyzerClusterState": 19, "shardsPerCollection": 0}
```


```
GET localhost:9200/_plugins/_performanceanalyzer/cluster/config?expand
{"currentPerformanceAnalyzerClusterState":{"PerformanceAnalyzerEnabled":true,"RcaEnabled":true,"LoggingEnabled":false,"BatchMetricsEnabled":false,"ThreadContentionMonitoringEnabled":false},"shardsPerCollection":0,"batchMetricsRetentionPeriodMinutes":7}
```

**Comment by dzane17** [comment]
TODO: update README api samples

File: src/main/java/org/opensearch/performanceanalyzer/config/setting/handler/PerformanceAnalyzerClusterSettingHandler.java (hunk 4 above)
Comment (dzane17) [reply src/main/java/org/opensearch/performanceanalyzer/config/setting/handler/PerformanceAnalyzerClusterSettingHandler.java:143]: The result of `getCurrentClusterSettingValue()` is only used to print cluster status. It is directly passed to the `XContentBuilder` so there is no need to change anything from the caller side.

Also, we decided to keep `getCurrentClusterSettingValue()` unchanged and add a separate function `getCurrentClusterSettingValueVerbose()` so UTs, ITs are not affected.

**Comment by dzane17** [comment]
> As this is a breaking change(as api response is essentially changing), I think we should instead introduce a new parameter like `?expand` and return meaningful status only when this is used. Using this we won't essentially need to fix [integ tests](https://github.com/opensearch-project/performance-analyzer/blob/3ab57eb8802329ac8a11e9fd3e1dbd605c8a35e7/src/test/java/org/opensearch/performanceanalyzer/PerformanceAnalyzerIntegTestBase.java#L262) or make change changes in RCA.
> 
> We can just update the README for this param usage. @kiranprakash154 @dzane17 Thoughts?
> 
> ```
> GET localhost:9200/_plugins/_performanceanalyzer/cluster/config
> {"batchMetricsRetentionPeriodMinutes": 7, "currentPerformanceAnalyzerClusterState": 19, "shardsPerCollection": 0}
> ```
> 
> ```
> GET localhost:9200/_plugins/_performanceanalyzer/cluster/config?expand
> {"currentPerformanceAnalyzerClusterState":{"PerformanceAnalyzerEnabled":true,"RcaEnabled":true,"LoggingEnabled":false,"BatchMetricsEnabled":false,"ThreadContentionMonitoringEnabled":false},"shardsPerCollection":0,"batchMetricsRetentionPeriodMinutes":7}
> ```

I have updated to only return detailed output when `?verbose` query param is passed

File: src/main/java/org/opensearch/performanceanalyzer/http_action/config/PerformanceAnalyzerClusterConfigAction.java [hunk 5]
```diff
@@ -160,6 +160,7 @@ public List<ReplacedRoute> replacedRoutes() {
     @Override
     protected RestChannelConsumer prepareRequest(final RestRequest request, final NodeClient client)
             throws IOException {
+        request.param("verbose");
```
Comment (sgup432) [inline src/main/java/org/opensearch/performanceanalyzer/http_action/config/PerformanceAnalyzerClusterConfigAction.java:163]: Why is this needed?

**Comment by sgup432** [comment]
Overall LGTM. Just one minor comment.

File: src/main/java/org/opensearch/performanceanalyzer/http_action/config/PerformanceAnalyzerClusterConfigAction.java (hunk 5 above)
Comment (dzane17) [reply src/main/java/org/opensearch/performanceanalyzer/http_action/config/PerformanceAnalyzerClusterConfigAction.java:163]: It is required by the RestRequest class to mark "verbose" as a consumed parameter. Else it rejects all unknown params.

Comment (dzane17) [reply src/main/java/org/opensearch/performanceanalyzer/http_action/config/PerformanceAnalyzerClusterConfigAction.java:163]: See comment on L234 - https://github.com/opensearch-project/performance-analyzer/blob/066cef88b29b5ddfa108ce8520f3b45436ac0866/src/main/java/org/opensearch/performanceanalyzer/http_action/config/PerformanceAnalyzerResourceProvider.java#L230-L238

Yes, this method is primarily meant to fetch the value of the `verbose` param which correspondingly marks it as "consumed." Since we do not care about the actual param value, there is no need to store the output of this call.

**Review by sgup432 — APPROVED** [review]

**Review by kkhatua — APPROVED** [review]
LGTM. Update the Description to check off the addition of Tests to your feature.

Would also be good to know why the JDK 11 and 17 builds are failing

**Review by kiranprakash154 — APPROVED** [review]
